TR123/Component.cs

```diff
@@ -118,33 +122,82 @@ public override void SetSettings(XmlNode settings)
     {
-        int layoutComponentsCount = state.Layout.LayoutComponents.Count;
-        if (_aslComponentPresent is null || layoutComponentsCount != _layoutComponentCount)
+        bool valuesNotInitialized = _aslComponentPresent is null;
+
+        var currentTimingMethod = state.CurrentTimingMethod;
+        var layoutAndTimingMethods = state
+            .Layout
+            .LayoutComponents
+            .Select(static comp =>
+                {
+                    return comp.Component switch
+                    {
+                        Timer timer => timer.Settings.TimingMethod,
+                        DetailedTimer detailedTimer => detailedTimer.Settings.TimingMethod,
+                        _ => comp.Component.ComponentName,
+                    };
+                }
+            )
+            .ToList();
+        bool layoutOrTimingMethodChanged = currentTimingMethod != _lsCurrentTimingMethod || !layoutAndTimingMethods.SequenceEqual(_layoutAndTimingMethods);
+
+        bool importantLayoutOrSettingChanged = valuesNotInitialized || layoutOrTimingMethodChanged;
+        if (importantLayoutOrSettingChanged)
         {
-            _layoutComponentCount = layoutComponentsCount;
-            HandleLayoutUpdates(state);
+            _layoutAndTimingMethods = layoutAndTimingMethods;
+            _lsCurrentTimingMethod = currentTimingMethod;
+            HandleLayoutOrSettingUpdates(state);
+
+            if (!ComponentSettings.GameVersionInitialized)
+            {
+                GameData.GameProcess = null;
+                GameData.GameVersion = GameVersion.None;
+            }
         }
 
         if (GameData.Update())
             base.Update(invalidator, state, width, height, mode);
     }
 
-    private void HandleLayoutUpdates(LiveSplitState state)
+    private void HandleLayoutOrSettingUpdates(LiveSplitState state)
     {
+        bool lsTimingMethodIsGameTime = _lsCurrentTimingMethod == TimingMethod.GameTime;
+        var timerWithGameTimeInLayout = false;
+        foreach (var timer in state.Layout.LayoutComponents.Where(static comp => comp.Component is Timer or DetailedTimer))
+        {
+            const string current = "Current Timing Method";
+            const string gameTime = "Game Time";
+            switch (timer.Component)
+            {
+                case DetailedTimer detailedTimerComponent:
+                    string method = detailedTimerComponent.Settings.TimingMethod;
+                    timerWithGameTimeInLayout = method == gameTime || (lsTimingMethodIsGameTime && method == current);
+                    break;
+                case Timer timerComponent:
+                    string timerMethod = timerComponent.Settings.TimingMethod;
+                    timerWithGameTimeInLayout = timerMethod == gameTime || (lsTimingMethodIsGameTime && timerMethod == current);
+                    break;
+            }
```

Comment:
This looks DRYable if the timing methods for each timer have the same type.

Reply:
`Timer` and `DetailedTimer` inherit from `IComponent`; otherwise, their types and even their inner Settings types differ (`TimerSettings` and `DetailedTimerSettings`).

Reply:
TimingMethod, however, is just a string. This was DRYed as a result.